thirdparty/CMakeLists.txt

```diff
@@ -25,14 +27,16 @@ set(CUB_INCLUDE_DIR "${cub_SOURCE_DIR}" PARENT_SCOPE)
 ###################################################################################################
 # - thrust ----------------------------------------------------------------------------------------
 
-message("Fetching Thrust")
+set(SRC_THRUST "https://github.com/thrust/thrust.git")
+
+message(STATUS "Fetching thrust from ${SRC_THRUST}")
 
 FetchContent_Declare(
     thrust
-    GIT_REPOSITORY https://github.com/thrust/thrust.git
+    GIT_REPOSITORY ${SRC_THRUST}
     GIT_TAG        1.9.10
     GIT_SHALLOW    true
-    PATCH_COMMAND COMMAND patch -p1 < "${CMAKE_CURRENT_SOURCE_DIR}/thrust-ret-if-fail.patch"
+    PATCH_COMMAND COMMAND patch -p1 < "${CMAKE_CURRENT_SOURCE_DIR}/thrust-ret-if-fail.patch" || true
```

Comment:
Adding `|| true` guarantees that the `patch` command always returns a `0` exit code. Previously, when `cmake ..` was run multiple times, successive calls to `patch` on already patched files would produce a nonzero exit code, and configuration would fail.

Reply:
This patch is going away soon hopefully.